shared_model/cryptography/ed25519_sha3_impl/crypto_provider.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_CRYPTOPROVIDER_HPP
+#define IROHA_CRYPTOPROVIDER_HPP
+
+#include "cryptography/ed25519_sha3_impl/signer.hpp"
+#include "cryptography/ed25519_sha3_impl/verifier.hpp"
+#include "cryptography/seed.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    /**
+     * Wrapper class for signing-related stuff.
+     */
+    class CryptoProvider {
+     public:
+      /**
+       * Signs the message.
+       * @param blob - blob to sign
+       * @param keypair - keypair
+       * @return Signed object with signed data
+       */
+      Signed sign(const Blob &blob, const Keypair &keypair) const;
```

Comment:
Low-level API of ed25519 library requires both keys for signing. Probably it could be reworked to deduce public key from a private one but I don't think this PR is a good place to do this (and I think we shouldn't change ed25519 internals as far as we absolutely have no other choice).